Modix.Services/Core/CoreSetup.cs

```diff
@@ -29,6 +28,9 @@ public static IServiceCollection AddModixCore(this IServiceCollection services)
                 .AddSingleton<IReadySynchronizationProvider>(x => x.GetService<ReadySynchronizationProvider>())
                 .AddSingleton<INotificationHandler<ReadyNotification>>(x => x.GetService<ReadySynchronizationProvider>())
                 .AddSingleton<ISelfUserProvider, SelfUserProvider>()
+                .AddSingleton<AuthorizationAutoConfigBehavior>()
+                .AddSingleton<INotificationHandler<GuildAvailableNotification>>(x => x.GetService<AuthorizationAutoConfigBehavior>())
+                .AddSingleton<INotificationHandler<JoinedGuildNotification>>(x => x.GetService<AuthorizationAutoConfigBehavior>())
```

Comment:
I don't think we can register these as singletons, since `IAuthorizationService` is scoped.